plugins/action/synchronize.py

```diff
@@ -333,10 +333,8 @@ def run(self, tmp=None, task_vars=None):
                     user = task_vars.get('ansible_ssh_user') or self._play_context.remote_user
 
             # Private key handling
-            private_key = self._play_context.private_key_file
-
-            if private_key is not None:
-                _tmp_args['private_key'] = private_key
+            # Use the private_key parameter if passed else use context private_key_file
```

Comment:
Could you please add a changelog entry for this change [here](https://github.com/ansible-collections/ansible.posix/tree/main/changelogs/fragments), add a test [here](https://github.com/ansible-collections/ansible.posix/blob/main/tests/integration/targets/synchronize/tasks/main.yml)?

Reply:
@Akasurde I will try to add the tests.
May I know why the build failed, it failed during yum install of gcc.

Reply:
Might be a temporary failure, I re-triggered CI job let us see.